src/rp2_common/hardware_sync/include/hardware/sync.h

```diff
@@ -299,7 +325,21 @@ spin_lock_t *spin_lock_init(uint lock_num);
  */
 void spin_locks_reset(void);
 
-// this number is not claimed
+/*! \brief Return a spin lock number from the _striped_ range
+ *  \ingroup hardware_sync
+ *
+ * Returns a spin lock number in the range PICO_SPINLOCK_ID_STRIPED_FIRST to PICO_SPINLOCK_ID_STRIPED_LAST
+ * in a round robin fashion. This does not grant the caller exclusive access to the spin lock, so the caller
+ * must:
+ *
+ * -# Abide (with other callers) by the contract of only holding this spin lock briefly (and with IRQs disabled - the default via \ref spin_lock_blocking()),
+ * and not whilst holding other spin locks.
+ * -# Be OK with any contention caused by the - brief due to the above requirement - contention with other possible users of the spin lock.
+ *
+ * \return lock_num a spin lock number the caller may use (non exclusively)
```

Comment:
I don't think you need the `lock_num` here? e.g. line 319 just says:
```
* \return The spin lock instance
```

Reply:
i'm happy with it